packages/matter.js/InterfaceNotes.ts

```diff
@@ -0,0 +1,121 @@
+//
+// Question 1. We just call this "Device" but it's really a "ServerDevice".
+// In a controller we could potentially also have a "ClientDevice" that looks
+// very similar but proxies over the fabric to a remote endpoint.  So should
+// we consider making space in the API for a future Client/Server bifurcation
+// or am I thinking about this wrong?
+//
+// Note 2. The concept of a "ClientDevice" wouldn't be just for controllers, it
+// would also be for client clusters of a "ServerDevice".  I think.
+
+// ATTRIBUTES VALUES
+//
+// Attribute values are stored in an immutable "State" object.  This is not
+// necessarily the "source of truth".  It's just what the implementation
+// has conveyed to matter.js as the current attribute values.
+//
+// Note 3: (no longer relevant)
+//
+// Question 2: (resolved myself)
+type State = {
+    readonly currentLevel: number | undefined;
+    readonly minLevel?: number;
+    readonly maxLevel?: number;
+    readonly onOffTransitionTime?: number;
+    readonly onLevel: number | undefined;
+    readonly onTransitionTime?: number | undefined;
+    readonly offTransitionTime?: number | undefined;
+    readonly defaultMoveRate?: number | undefined;
+    readonly options: number;
+    readonly remainingTime?: number;
+    readonly startUpCurrentLevel?: number | undefined;
+}
+
+export interface PumpDevice extends Device {
+    // ATTRIBUTE MUTATION
+    //
+    // This is how the device implementation tells matter.js that attribute
+    // values are different.  Internal implementation would then read all
+    // attribute values and push any changes into the appropriate
+    // ClusterServer.
+    //
+    // Question 3: One reason I like this approach is I worry that notifying
+    // matter.js for every change would get too chatty.  Right now changes
+    // are coalesced in SubscriptionHandler before notifying subscribers so
+    // I don't think this is an issue from a network perspective.  But there
+    // are a lot of internal function calls for every attribute on every change
+    // I don't think there's any immediate need for change but is it worth
+    // considering for the shape of the public API for when there are hundreds
+    // of endpoints and thousands of attributes on a low-power device?
+    //
+    // Note 4: (no longer relevant)
+    set(changes: Partial<State>): void;
+
+    // This is a convenience form of set for updating a single attribute.
+    //
+    // Note 8: We could also generate a setter for every individual attribute.
+    // However, I worry that would encourage inefficient implementation.
+    //
+    // Note 9: Not sure how to validate value type in the interface so might
+    // need to do that programmatically.
+    set(name: keyof (State), value: any): void;
+
+    // This is how matter.js conveys a request to change values from the
+    // fabric.  It would be wired to AttributeServers via InteractionServer
+    // internally.
```

Comment:
OK.  I'll see how that affects wiring.  Maybe will want separate interfaces for "configuration" vs "session" attributes.

Also, r.e. earlier discussion about the need to propagate and/or veto writes before commit...  That only applies to configuration attributes.  So might be another benefit to separate interfaces.

Reply:
But in fact it only affects wiring locally by adding afabric parametrr unless someone finds a better idea. The question is also if it will be really an "local writing topic" for these. will wait for use cases where devs have issues with that approach :) But there will also be a way to get Fabrics - then it is "just choosing" if ther eis more then one